prov/rxm/src/rxm_msg.c

```diff
@@ -802,17 +802,17 @@ rxm_send_common(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_conn,
 		(data_len > rxm_ep->rxm_info->tx_attr->inject_size)) ||
 	       (data_len <= rxm_ep->rxm_info->tx_attr->inject_size));
 
-	if (data_len <= rxm_ep->eager_limit) {
+	iface = rxm_mr_desc_to_hmem_iface_dev(desc, count, &device);
+
+	if (data_len <= rxm_ep->eager_limit && iface != FI_HMEM_ZE) {
 		ret = rxm_send_eager(rxm_ep, rxm_conn, iov, desc, count,
 				     context, data, flags, tag, op,
 				     data_len, total_len);
-	} else if (data_len <= rxm_ep->sar_limit) {
+	} else if (data_len <= rxm_ep->sar_limit && iface != FI_HMEM_ZE) {
```

Comment:
Can we restructure the checks to avoid comparing iface against ZE multiple times?  I don't know if we want to move the rndv check to the top or use a goto to jump down.  Eager might be most common for MPI, but rndv may be for oneCCL.

Reply:
I vote for `goto`. That introduces the least changes to the original flow.